www/js/control/uploadService.js

```diff
@@ -94,46 +94,62 @@ angular.module('emission.services.upload', ['emission.plugin.logger'])
             newScope.fromDirText = $translate.instant('upload-service.upload-from-dir',  {parentDir: parentDir});
             newScope.toServerText = $translate.instant('upload-service.upload-to-server',  {serverURL: uploadConfig});
 
+            var didCancel = true;
+
             const detailsPopup = $ionicPopup.show({
                 title: $translate.instant("upload-service.upload-database", { db: database }),
                 template: newScope.toServerText
                     + '<input type="text" ng-model="data.reason"'
                     +' placeholder="{{ \'upload-service.please-fill-in-what-is-wrong \' | translate}}">',
                 scope: newScope,
                 buttons: [
-                  { text: 'Cancel' },
+                  { 
+                    text: 'Cancel',
+                    onTap: function(e) {
+                        didCancel = true;
+                        detailsPopup.close();
+                    }
+                  },
                   {
                     text: '<b>Upload</b>',
                     type: 'button-positive',
                     onTap: function(e) {
-                      if (!newScope.data.reason) {
+                        didCancel = false;
+                        if (!newScope.data.reason) {
                         //don't allow the user to close unless he enters wifi password
-                        e.preventDefault();
-                      } else {
-                        return newScope.data.reason;
-                      }
+                            e.preventDefault();
+                        } else {
+                            return newScope.data.reason;
+                        }
```

Comment:
nit: Can you remove the whitespace changes so that we don't get incorrect history/blame for the commits?